lib/cypress/create_download_zip.rb

```diff
@@ -30,12 +30,24 @@ def self.all_patients
       file
     end
 
-    def self.create_total_test_zip(product, format = 'qrda')
-      measure_tests = MeasureTest.where(product_id: product.id)
+    def self.create_total_test_zip(product, format = 'qrda', criteria_list)
+      product_tests = ProductTest.where(product_id: product.id)
       file = Tempfile.new("all-patients-#{Time.now.to_i}")
       Zip::ZipOutputStream.open(file.path) do |z|
-        measure_tests.each do |m|
-          add_file_to_zip(z, "#{m.cms_id}_#{m.id}.#{format}.zip".tr(' ', '_'), m.patient_archive.read)
+        added_filter_test = false
+        product_tests.each do |pt|
+          if pt.class == MeasureTest
+            add_file_to_zip(z, "#{pt.cms_id}_#{pt.id}.#{format}.zip".tr(' ', '_'), pt.patient_archive.read)
+          elsif pt.class == FilteringTest
+            add_file_to_zip(z, "filteringtest_#{pt.cms_id}_#{pt.id}.#{format}.zip".tr(' ', '_'), pt.patient_archive.read) if !added_filter_test
+            added_filter_test = true
+          elsif pt.class == ChecklistTest
+            @product = pt.product
+            zip = Cypress::CreateDownloadZip.create_c1_criteria_zip(@product.product_tests.checklist_tests.first, criteria_list).read
+            add_file_to_zip(z, "checklisttest_#{@product.name}_#{@product.id}_c1_manual_criteria.zip".tr(' ', '_'),zip)
+          else
+            raise "Unknown product test type in download: " >> pt._type
```

Comment:
If you're concatenating strings in Ruby, you'll want to use `+` rather than `<<`